### PR TITLE
Updates and improvements for node-sass v3 (polished)

### LINKED ISSUE #22: Doesn't log to console when sass file is invalid
There is no callback for checking errors nor logs to console. So, it is a bit hard to identify problem.
Maybe it's because I use it without connect module. 

```javascript
// middleware.js
module.exports = function(options, cb){ // line 44

if (err) { cb(err); return next(err); } // line 115

// when using
app.use('/stylesheets', sassMiddleware({
  src: "./public/stylesheets/",
  debug: true,
  outputStyle: 'expanded'
}, function(err){
   console.log(err);
}));
```

### LINKED ISSUE #31: If an imported file is not found request never ends
If an imported file doesn't exist the site is not able to load the stylesheet and stays waiting indefinitely.
There are no notifications in console.

image

### PR DESCRIPTION
**Pull request by bumxu**
- Use of strict mode. :cop: 
- Updates and some bug fix for node-sass v3.
- Minor changes in formatting and documentation.
 - Removed unused/unsupported code, etc.
 - Additional comments.
 - Documented some undocumented options.
- This should fix #22 with the formatted error output in debug mode and the call to `error` function.
- This should fix #31 with a natural flow `next(err) --> use(function(err, req, res, next){})`.
- About #19, it doesn't end the connection nevermore, successive middlewares can add headers.
- Updates in `error` handling ([summary](https://github.com/bumxu/node-sass-middleware/commit/ca9495761c8349f59bc350af0e7c0e2050dc3f6a)).
- Tests revision for v3 and error handling.

It includes commit iMoritz@1ea16c1 by @iMoritz fork. :punch: 

Travis is now happy again. :smile:

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -28,55 +34,65 @@ var imports = {};
...
-  // Debug option
-  var debug = options.debug;
-
-  // Source dir required
-  var src = options.src;
-  if (!src) { throw new Error('sass.middleware() requires "src" directory'); }
+  // This function will be called if something goes wrong
+  var error = typeof options.error === 'function' ? options.error : function (err) {
+    throw err;
+  }
```
Comment (am11) [inline middleware.js]: `};`

File: middleware.js [hunk 2]
```diff
@@ -28,55 +34,65 @@ var imports = {};
...
 
-  // Default dest dir to source
+  // Source directory (required)
+  var src = options.src || function() {
+    throw new Error('sass.middleware() requires "src" directory.');
+  }();
```
Comment (am11) [inline middleware.js]: This is RAD! I like it! 
@kevva, what do you think about this?

**Comment by am11** [comment]
It looks good to me! :+1: 
Would you like to add that missing test you mentioned before this get merged?

**Comment by kevva** [comment]
I'll review it later today.

**Comment by bumxu** [comment]
@am11, a question about [this](https://github.com/sass/node-sass-middleware/pull/41#discussion-diff-29997234), I didn't finish understand...
Why `error` function now throws an error instead of calling `next()`?
I need understand it to make Travis happy, thanks! :yum:

**Comment by am11** [comment]
@bumxu, this should do the trick: https://gist.github.com/am11/66d4b708c15159f94dbc.

Side note: `nodeSass.render()` *returns* the `Error` object and `nodeSass.renderSync()` *throws* it.

**Comment by bumxu** [comment]
@am11 Great! :+1:

File: package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
   },
   "dependencies": {
     "mkdirp": "0.5.x",
-    "node-sass": "^2.0.1"
+    "node-sass": "^3.0.0"
   },
   "devDependencies": {
     "connect": "^3.3.4",
```
Comment (am11) [inline package.json]: Please replace with:

```json
  "dependencies": {
    "mkdirp": "^0.5.0",
    "node-sass": "^3.1.0"
  },
  "devDependencies": {
    "connect": "^3.3.5",
    "mocha": "^2.2.4",
    "should": "^6.0.1",
    "supertest": "^1.0.1"
  }
```

Comment (bumxu) [inline package.json]: :+1:

**Comment by am11** [comment]
:tada: 
Thanks for your contribution. This one is HUGE! :sunny:

**Pull request merged**